### PR TITLE
Implement constructor MockTextBuffer(string)

### PR DESCRIPTION
**Pull request by hoanhtien**
This fixes https://github.com/Microsoft/nodejstools/issues/413.

Most TestComment* test cases use the following constructor of MockTextBuffer which by some reason is not implemented :neutral_face:
```C#
MockTextBuffer.cs
31  public MockTextBuffer(string content) {
32  }
```

All TestComment* pass after this change.

### CODE REVIEW

File: Common/Tests/Utilities/Mocks/MockTextBuffer.cs [hunk 1]
```diff
@@ -28,14 +28,14 @@ public class MockTextBuffer : ITextBuffer {
         /// </summary>
         private PropertyCollection _properties;
 
-        public MockTextBuffer(string content) {
+        public MockTextBuffer(string content) : this(content, "Node.js") {
```
Comment (paulvanbrenk) [inline Common/Tests/Utilities/Mocks/MockTextBuffer.cs]: Can you use the contentType constant for this?

**Comment by mousetraps** [comment]
This change is in the common codebase that we share with Python Tools, among others. We'll need to find a way to make it work without adding Node.js-specific code to Common/*

**Comment by hoanhtien** [comment]
Derived TestUtilities.Mocks.MockTextBuffer to avoid modifying it directly. Thank Paul for this solution.

File: Nodejs/Tests/Core/CommentBlockTests.cs [hunk 2]
```diff
@@ -17,7 +17,8 @@
 using Microsoft.NodejsTools.Editor.Core;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Microsoft.VisualStudio.Text;
-using TestUtilities.Mocks;
+using MockTextView = TestUtilities.Mocks.MockTextView;
```
Comment (paulvanbrenk) [inline Nodejs/Tests/Core/CommentBlockTests.cs]: Make sure this works in Dev12

File: Nodejs/Tests/Core/Mocks/MockTextBuffer.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿using Microsoft.NodejsTools;
+
+namespace NodejsTests.Mocks {
+    class MockTextBuffer : TestUtilities.Mocks.MockTextBuffer {
```
Comment (paulvanbrenk) [inline Nodejs/Tests/Core/Mocks/MockTextBuffer.cs]: Add a constructor which takes 2 arguments so people can specify the filename.

In the implementation generate the filename the same way the base class does when it's not specified, except with a js file.

File: Nodejs/Tests/Core/CommentBlockTests.cs (hunk 2 above)
Comment (hoanhtien) [inline Nodejs/Tests/Core/CommentBlockTests.cs]: Confirmed this works in Dev12.

File: Nodejs/Tests/Core/Mocks/MockTextBuffer.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿using System.IO;
```
Comment (mousetraps) [inline Nodejs/Tests/Core/Mocks/MockTextBuffer.cs]: add copyright header

File: Nodejs/Tests/Core/Mocks/MockTextBuffer.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+namespace NodejsTests.Mocks {
+    class MockTextBuffer : TestUtilities.Mocks.MockTextBuffer {
+		public MockTextBuffer(string content) : base(content: content, contentType: NodejsConstants.Nodejs) {
+        }
+
+        private static string GetRandomFileNameIfNull(string filename) {
+            if (filename == null) {
+                filename = Path.Combine(TestUtilities.TestData.GetTempPath(), Path.GetRandomFileName(), "file.js");
+            }
+            return filename;
```
Comment (mousetraps) [inline Nodejs/Tests/Core/Mocks/MockTextBuffer.cs]: you can simplify this to `return filename ?? Path.Combine(...);`

**Comment by mousetraps** [comment]
minor comments, otherwise lgtm

**Comment by mousetraps** [comment]
did you get a chance to take a look at these comments yet? should be quick enough to resolve

**Comment by mjbvz** [comment]
See updated PR #826 based on this one. Closing this PR.

**Pull request closed without merging**